### ISSUE TITLE
Cannot download youtube video anymore

### ISSUE THREAD

**Issue by Atomicboot** [monkeyWie/gopeed-extension-youtube#12]
Every youtube video i try to download find only a file "watch" without any extension.
![A](https://github.com/user-attachments/assets/854483d3-eb52-4c43-a3fb-c5398dfbd8fa)
![B](https://github.com/user-attachments/assets/e4428594-831e-477d-a77c-fdab525432bc)

The file downloaded cannot be open even adding an extension manually.

**Comment by monkeyWie** [comment]
what's your gopeed version?

**Comment by Atomicboot** [comment]
1.5.7.
I just updated to 1.6.5, but i'm getting this errors after i click on confirm:

![image](https://github.com/user-attachments/assets/73427b2e-a615-408e-aabc-319080993c04)
![image](https://github.com/user-attachments/assets/7464fd7d-6e00-461b-aca4-d69f7f25c932)

**Comment by monkeyWie** [comment]
It seems there is an issue with your network, please make sure your network can directly access YouTube

**Comment by Atomicboot** [comment]
Ye i confirm that my network can access correctly to youtube.

I paste here my container settings:

![image](https://github.com/user-attachments/assets/55e28b37-a821-4f80-94a5-abc47d680780)
![image](https://github.com/user-attachments/assets/0a53faed-e8f6-4c32-84d9-4f9a7763a0d1)
![image](https://github.com/user-attachments/assets/9b4bbce5-a094-408d-b0d9-36fc50d2ca19)

**Comment by monkeyWie** [comment]
![image](https://github.com/user-attachments/assets/e4c38cf9-f778-4a84-8760-cb4bbf50e52d)

I test it work fine, are you downloaded youtube video before that?

**Comment by Atomicboot** [comment]
I didn't downloaded a youtube video before, but is really strange, cause i've correctly access to youtube, i even just ping youtube correctly.

![image](https://github.com/user-attachments/assets/dd283238-327a-4388-88e3-eedb5a817829)

**Comment by X-49** [comment]
+1 the same problem. YouTube is blocked in Russia. If you don't set “Network --> Proxy --> System Proxy” in Gopeed, this extension can't download anything from YouTube, but with "System Proxy" set, it simply downloads an empty file. No problem with my VPN connection, using FlClash client with AmneziaWG configuration.

**Comment by monkeyWie** [comment]
@X-49 Have you tried the latest extension version?

**Comment by X-49** [comment]
> @X-49 Have you tried the latest extension version?

Yes, I have the latest version of Gopeed installed and am using the latest version of your extension. Everything was installed today, a few hours ago.

**Comment by monkeyWie** [comment]
As far as I know, YouTube will block some IPs from watching videos. You can open YouTube video in a private browser window to verify if the IP is blocked. If it requires login, it means the IP is blocked and you need to set a cookie to download.

**Comment by X-49** [comment]
> As far as I know, YouTube will block some IPs from watching videos. You can open YouTube video in a private browser window to verify if the IP is blocked. If it requires login, it means the IP is blocked and you need to set a cookie to download.

I periodically use the private mode in the browser and no, such a problem has never noticed, any videos are quietly played without the requirement to log in. If anything, my AmneziaWG configuration uses Proton VPN (Netherlands), which is wrapped in the same config file in Cloudflare WARP, i.e. my endpoint — is “Amsterdam, Netherlands” ISP Cloudflare. With such a VPN connection my web surfing is much more pleasant than with native IPv4, captcha is practically non-existent.

I think it's clear now that the problem is not with the VPN?

**Comment by monkeyWie** [comment]
Alright,  can you test downloading videos directly using Node.js with https://github.com/LuanRT/YouTube.js?

**Comment by X-49** [comment]
> Alright, can you test downloading videos directly using Node.js with https://github.com/LuanRT/YouTube.js?

Sorry, but I don't quite understand how to interact with this JS client. I can only add that I also checked VLESS configuration in Xray-core in System Proxy and TUN mode — the problem still remains. Instead of video, a 1 MB “watch” file is downloaded.

**Comment by AuthorProxy** [comment]
just installed latest app and extension

<img width="832" height="526" alt="Image" src="https://github.com/user-attachments/assets/f79a3634-203a-4b88-95b6-004cc3509435" />

do not have any problems with other similar apps for youtube

**Comment by exe19890522** [comment]
https://www.youtube.com/watch?v=W3oCAMc-mAU   
when i click confirm to download it ,just only give me a file:
which in side is:<!DOCTYPE html><html style="font-size: 10px;font-family: Roboto, Arial, sans-serif;" lang="en" darker-dark-theme darker-dark-theme-deprecate system-icons typography typography-spacing><head> ....

**Comment by sherow1982** [comment]
same here

**Comment by Leroy-X** [comment]
Click Download and get an error unsupported protocol

**Comment by dh336699** [comment]
Met a same error until today as well.

**Comment by Bijman** [comment]
I have the same issue on Gopeed v1.9.3.
